src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,115 @@
 package com.theironyard.novauc;
 
+import spark.ModelAndView;
+import spark.Session;
+import spark.Spark;
+import spark.template.mustache.MustacheTemplateEngine;
+
+import java.util.HashMap;
+
 public class Main {
 
-    public static void main(String[] args) {
-	// write your code here
+
+
+    public static void main(String[] args){
+
+        HashMap<String, User> users = new HashMap<>();
+
+        Spark.staticFileLocation("templates");
+
+
+        Spark.init();
+        Spark.get(
+                "/",
+                ((request, response) -> {
+                    Session session = request.session();
+                    String name = session.attribute("userName");
+                    User user = users.get(name);
+
+                    HashMap m = new HashMap<>();
+                    if (user == null) {
+                        return new ModelAndView(m, "login.html");
+                    }
+                    else {
+                        return new ModelAndView(user, "home.html");
+                    }
+                }),
+                new MustacheTemplateEngine()
+        );
+        Spark.post(
+                "/login",
+                ((request, response) -> {
+                    String name = request.queryParams("loginName");
+                    User user = users.get(name);
+                    if (user == null) {
+                        user = new User(name);
+                        users.put(name, user);
+                    }
+
+                    Session session = request.session();
+                    session.attribute("userName", name);
+
+                    response.redirect("/");
+                    return "";
+                })
+        );
+        Spark.post(
```

Comment:
Good job Kevin.  In the future try to vary variable names a bit.  You have three things called messages, it made my head spin.